docs/publish/deploy-office-add-in-sso-to-azure.md

```diff
@@ -0,0 +1,253 @@
+     *
+     * This file is the main Node.js server file that defines the express middleware.
+     */
+    
+    require("dotenv").config();
+    import * as createError from "http-errors";
+    import * as path from "path";
+    import * as cookieParser from "cookie-parser";
+    import * as logger from "morgan";
+    import express from "express";
+    import { getUserData } from "./msgraph-helper";
+    import { validateJwt } from "./ssoauth-helper";
+    
+    /* global console, process, require, __dirname */
+    
+    const app = express();
+    const port = process.env.PORT;
+    
+    app.set("port", port);
+    
+    // view engine setup
+    app.set("views", path.join(__dirname, "views"));
+    app.set("view engine", "pug");
+    
+    app.use(logger("dev"));
+    app.use(express.json());
+    app.use(express.urlencoded({ extended: false }));
+    app.use(cookieParser());
+    
+    /* Turn off caching when developing */
+    if (process.env.NODE_ENV !== "production") {
+      app.use(express.static(path.join(process.cwd(), "dist"), { etag: false }));
+    
+      app.use(function (req, res, next) {
+        res.header("Cache-Control", "private, no-cache, no-store, must-revalidate");
+        res.header("Expires", "-1");
+        res.header("Pragma", "no-cache");
+        next();
+      });
+    } else {
+      // In production mode, let static files be cached.
+      app.use(express.static(path.join(process.cwd())));
+      console.log("static set up: " + path.join(process.cwd()));
+    }
+    
+    const indexRouter = express.Router();
+    indexRouter.get("/", function (req, res) {
+      //   res.render("/taskpane.html");
+      res.sendFile("/taskpane.html", { root: __dirname });
+    });
+    
+    app.use("/", indexRouter);
+    
+    app.get("/getuserdata", validateJwt, getUserData);
+    
+    // Catch 404 and forward to error handler
+    app.use(function (req, res, next) {
+      next(createError(404));
+    });
+    
+    // error handler
+    app.use(function (err, req, temp, res) {
+      // set locals, only providing error in development
+    
+      res.locals.message = err.message;
+      res.locals.error = req.app.get("env") === "development" ? err : {};
+    
+      // render the error page
+      res.status(err.status || 500).send({
+        message: err.message,
+      });
+    });
+    
+    app.listen(process.env.PORT, () => console.log("Server listening on port: " + process.env.PORT));
+    ```
+
+1. Save the file.
+
+## Update app registration
+
+We recommend you create multiple app registrations for localhost, staging, and deployment testing. The following steps ensure that the app registration you use for deployment correctly uses the app service URL.
+
+1. In the Azure portal, open your app registration. Note that the app registration may be in a different account than your app service. Be sure to sign in to the correct account.
+1. In the left sidebar, select **Authentication**.
```

Comment:
FWIW, I got confused here because the Azure page I was already on had an Authentication section. Not sure how to make it more clear to go elsewhere...

Reply:
I added an image which hopefully makes this clear.